src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/SmoDatabaseCustomNode.cs

```diff
@@ -14,14 +14,54 @@ namespace Microsoft.SqlTools.ServiceLayer.ObjectExplorer.SmoModel
     /// </summary>
     internal partial class DatabasesChildFactory : SmoChildFactoryBase
     {
+        private const string databaseUnavailableKey = "databaseUnavailable";
+
         public override string GetNodeStatus(object smoObject, SmoQueryContext smoContext)
         {
             return DatabasesCustomNodeHelper.GetStatus(smoObject, smoContext, CachedSmoProperties);
         }
+
+        public override Dictionary<string, object> GetExtraProperties(object smoObject, SmoQueryContext smoQueryContext)
+        {
+            var properties = new Dictionary<string, object>();
+            properties.Add(databaseUnavailableKey, DatabasesCustomNodeHelper.GetDatabaseIsUnavailable(smoObject, smoQueryContext, CachedSmoProperties));
+            return properties;
+        }
     }
 
     internal static class DatabasesCustomNodeHelper
     {
+        private static readonly DatabaseStatus[] unavailableDatabaseStatuses = { DatabaseStatus.Inaccessible, DatabaseStatus.Offline, DatabaseStatus.Recovering,
```

Comment:
Does this give a style warning? I thought static readonly fields were CapitalCase

Reply:
No style warning, and elsewhere in the code we're not consistent between starting them with an uppercase and lowercase letter (search 'private static readonly' if you're interested).

Is one of them preferred?